exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java

```diff
@@ -83,12 +83,17 @@ public QueryResult run(final WorkManager workManager, final WebUserConnection we
     float usagePercent = getHeapUsage();
 
     // Wait until the query execution is complete or there is error submitting the query
-    logger.debug("Wait until the query execution is complete or there is error submitting the query");
+    if (logger.isDebugEnabled()) {
+      logger.debug("Wait until the query execution is complete or there is error submitting the query");
+    }
     do {
       try {
-        isComplete = webUserConnection.await(TimeUnit.SECONDS.toMillis(1)); /*periodically timeout to check heap*/
-      } catch (Exception e) { }
-
+        isComplete = webUserConnection.timedWait(TimeUnit.SECONDS.toMillis(1)); /*periodically timeout 1sec to check heap */
```

Comment:
Actually you don't need this `timedWait` method. You can still use same `await(timeout)` as you were using earlier. But outside while loop you can do below check to get the exception in error condition. Check [here](https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/work/prepare/PreparedStatementProvider.java#L158) 

```
// check if query completed with exception
if (isComplete && webUserConnection.getError() != null) {
      throw new UserRemoteException(webUserConnection.getError());
}
```

Reply:
For some strange reason, the webUserConnection.getError() was not giving me the Exception message for invalid schemas. Let me see if I missed something.

Reply:
Yep. It's sufficient to just wrap the DrillPBError into the UserException. I'm getting the invalid schema exception reported as well.